### PR TITLE
Add config to instantiate ConsulSecretStore

### LINKED ISSUE #40: Consul backed Secret Store
We currently have an `InMemorySecretStore` that does well for mocking, but we need to store `Secrets` somewhere that provides a consistent view of the current state of `Secrets`.

Obviously, Zookeeper is a great place to store these things, but since we have a use-case for Consul as well.

All that needs to be done is to provide a Consul backend for a `SecretStoreApi` interface, making sure it is as purely functional as you can -- why not have the entire state of everything ever forever?

### PR DESCRIPTION
**Pull request by maheshkelkar**
- Also updated the README.md for secret and session store config
  examples
- Add version to fat jar

Fixes #40

### CODE REVIEW

File: server/src/test/scala/com/lookout/borderpatrol/server/ConfigSpec.scala [hunk 1]
```diff
@@ -70,13 +71,13 @@ class ConfigSpec extends BorderPatrolSuite {
   it should "uphold encoding/decoding ServerConfig" in {
     def encodeDecode(config: ServerConfig) : ServerConfig = {
       val encoded = config.asJson
+      println("***MVK: encoded = " + encoded)
```
Comment (trane) [inline server/src/test/scala/com/lookout/borderpatrol/server/ConfigSpec.scala]: println?

File: server/src/test/scala/com/lookout/borderpatrol/server/ConfigSpec.scala [hunk 2]
```diff
@@ -70,13 +71,13 @@ class ConfigSpec extends BorderPatrolSuite {
...
       decode[ServerConfig](encoded.toString()) match {
         case Xor.Right(a) => a
         case Xor.Left(b) => ServerConfig(defaultSecretStore, defaultSessionStore, Set(), Set(), Set(), Set())
       }
     }
-    val foo = encodeDecode(serverConfig)
-    verifyServerConfig(encodeDecode(serverConfig), serverConfig)
+    //verifyServerConfig(encodeDecode(serverConfig), serverConfig)
```
Comment (trane) [inline server/src/test/scala/com/lookout/borderpatrol/server/ConfigSpec.scala]: why comment this out?

Comment (maheshkelkar) [inline server/src/test/scala/com/lookout/borderpatrol/server/ConfigSpec.scala]: fixed. mistake

File: server/src/test/scala/com/lookout/borderpatrol/server/ConfigSpec.scala (hunk 1 above)
Comment (maheshkelkar) [inline server/src/test/scala/com/lookout/borderpatrol/server/ConfigSpec.scala]: debugging. took it out

**Pull request merged**